src/lib/pages/frontsite/Documents/DocumentDetails/DocumentDetails.js

```diff
@@ -126,20 +126,25 @@ DocumentDetailsLayout.defaultProps = {
 
 class DocumentDetails extends Component {
   componentDidMount() {
-    const { match } = this.props;
-    this.fetchDocumentsDetails(match.params.documentPid);
+    const { documentDetails, match } = this.props;
+    if (_isEmpty(documentDetails.metadata)) {
```

Comment:
In which cases the document metadata is already provided via a prop?
Is it happening always?

Reply:
we exported this component as not redux controlled, to be able to use it independently. In this case the data will be passed from ownProps instead of the redux mapping. We are using it to save one frontend redirect when trying to access record via legacy_recid.